src/3pc/agent/runners/evm/impl.cpp

```diff
@@ -155,10 +155,35 @@ namespace cbdc::threepc::agent::runner {
         }
         auto tn = to_uint64(maybe_tn.value());
         auto tn_key = m_host->ticket_number_key(tn);
+        auto cb
```

Comment:
Note: I am not fully sure if this is what was meant by the comment to "split the nested lambdas into separate functions" -- would be good to have a second opinion

Reply:
Thanks @maurermi . May I also suggest we replace the new `cb()` another private member function as follows:

`evm_runner::lock_tx_receipt( const cbdc::buffer& valid_txid ) { ... }`

Let me now what you think.

Reply:
@AlexRamRam I think that is a fair change, would help with readability

Reply:
@AlexRamRam I realized that this may not be necessarily feasible because `cb` requires `tn` to be in its scope. That would indicate that if it were made a separate method, it would require tn to be passed as an argument or for `tn` to be stored at class or global scope. It cannot be passed as an argument because `m_try_lock_callback` expects a function that takes one argument of type `const broker::interface::try_lock_return_type&`. I think rather than storing `tn` in a broader scope, perhaps it's simpler to just rename `cb` to a more informative name, (ex. `handle_tx_lock_receipt_callback` ?)